test/features/weather/data/weather_data_source_test.dart

```diff
@@ -0,0 +1,105 @@
+import 'dart:convert';
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:flutter_training/features/weather/data/weather_data_source.dart';
+import 'package:flutter_training/features/weather/model/fetch_weather_request.dart';
+import 'package:flutter_training/features/weather/model/fetch_weather_response.dart';
+import 'package:flutter_training/features/weather/model/weather_condition.dart';
+import 'package:mockito/annotations.dart';
+import 'package:mockito/mockito.dart';
+import 'package:yumemi_weather/yumemi_weather.dart';
+
+@GenerateNiceMocks(
+  [
+    MockSpec<YumemiWeather>(),
+  ],
+)
+import 'weather_data_source_test.mocks.dart';
+
+void main() {
+  final api = MockYumemiWeather();
+  final client = WeatherDataSource(api);
```

Comment:
[imo]
個人的には `dataSource` のほうが自然かなと思いました。
```suggestion
  final dataSource = WeatherDataSource(api);
```

Reply:
ありがとうございます 修正しました
56eb59d7a9933ae853d605fce88aab197398f20b